### DISCUSSION TITLE
[@types/youtube] Add getVideoData() and possibly other missing functions

### DISCUSSION THREAD

**Discussion by jericjan** [DefinitelyTyped/DefinitelyTyped#69298] [Issues with a @types package]
Strangely enough, `getVideoData()` used to exist, as seen in PR #2575. But at some point, it got removed somewhere in the transition from `youtube.d.ts` to `index.d.ts`

To add to that, the `YT.Player` apparently has 200 functions that can be seen by running this on any YouTube page:
```js
Object.getOwnPropertyNames(document.querySelector(".html5-video-player"));
```
Meanwhile the type definition file currently has 42. I don't actually need them all added. I just thought I'd let you know.

[a comment by an automated account is not shown]

**Comment by moloko** [comment]
Additionally, there's some functionality on the `window.YT` object itself that could be useful:
<img width="727" height="248" alt="YT-API" src="https://github.com/user-attachments/assets/cfcd547f-2691-4645-98d9-6d3efe5402c6" />

For example, I've seen the `YT.ready` function used [like this](https://github.com/videojs/videojs-youtube/blob/050c1ee92e4dfd108dbe5bbe0ec3da6decf141a5/src/Youtube.js#L729-L737) as an alternative to the (very old-fashioned) `window.onYouTubeIframeAPIReady` way of getting a callback for the YT iFrame API being ready...
